astro/utils/parallel.go

```diff
@@ -17,15 +17,27 @@
 package utils
 
 import (
+	"context"
 	"sync"
 )
 
 // Parallel runs at most maxConcurrent functions in parallel.
-func Parallel(maxConcurrent int, fns ...func()) {
+func Parallel(ctx context.Context, maxConcurrent int, fns ...func()) {
 	wg := sync.WaitGroup{}
 	guard := make(chan struct{}, maxConcurrent)
+
+	isCancelled := false
```

Comment:
Consider using [x/sync/sempaphore](https://godoc.org/golang.org/x/sync/semaphore) instead, which supports exactly this pattern, with cancellation

Reply:
thanks for your suggestion @jcorbin, I've added a commit addressing it